### ISSUE TITLE
Compatibility with Symfony 4.2

### ISSUE THREAD

**Issue by QuentinCurtet** [KnpLabs/KnpMenuBundle#395]
Upgrading to symfony 4.2 with this bundle leads to this deprecation :
> A tree builder without a root node is deprecated since Symfony 4.2 and will not be supported anymore in 5.0.

It is precisely in this file : 
https://github.com/KnpLabs/KnpMenuBundle/blob/master/src/DependencyInjection/Configuration.php

**Comment by althaus** [comment]
New version'd be:

~~~php
    $treeBuilder = new TreeBuilder('knp_menu');
    $rootNode = $treeBuilder->getRootNode();
~~~

I guess this'd need a Symfony version check to use one or the other style.

**Comment by QuentinCurtet** [comment]
Yes i think
I don't know if it could cause BC breaks.

But i think this could be interesting to PR your solution, can you do this ?

**Comment by umulmrum** [comment]
I built a possible solution in https://github.com/KnpLabs/KnpMenuBundle/pull/396

**Comment by garak** [comment]
This has been fixed with merge of #399 
This issue should be closed
